lib/src/tag_detection.dart

```diff
@@ -501,15 +506,29 @@ class _DeclaredFlutterPlatformDetector {
     return _declaredPlatformCache.putIfAbsent(packageName, () {
       final result = <_FlutterPlatform>{};
       final fields = _pubspecCache.pubspecOfPackage(packageName).toJson();
-      if (fields['flutter'] is! Map ||
-          fields['flutter']['plugin'] is! Map ||
-          fields['flutter']['plugin']['platforms'] is! Map) {
+      if (fields['flutter'] is! Map || fields['flutter']['plugin'] is! Map) {
+        // If a package doesn't declare support for any platforms, it is
+        // not a plugin, and can work on any platforms compatible with
+        // imported constraints.
         return _FlutterPlatform.recognizedPlatforms.toSet();
       }
-      final declaredPlatforms = fields['flutter']['plugin']['platforms'] as Map;
-      for (final platform in _FlutterPlatform.recognizedPlatforms) {
-        if (declaredPlatforms.containsKey(platform.name)) {
-          result.add(platform);
+      final pluginMap = fields['flutter']['plugin'] as Map;
+
+      if (pluginMap['platforms'] is Map) {
+        final declaredPlatforms =
+            fields['flutter']['plugin']['platforms'] as Map;
+        for (final platform in _FlutterPlatform.recognizedPlatforms) {
+          if (declaredPlatforms.containsKey(platform.name)) {
+            result.add(platform);
+          }
+        }
+      } else {
+        // Legacy style plugin:
+        if (pluginMap['androidPackage'] is String) {
+          result.add(_FlutterPlatform.android);
+        }
+        if (pluginMap['iosPrefix'] is String) {
+          result.add(_FlutterPlatform.ios);
         }
```

Comment:
The old plugin format didn't support either ios or android, but required both to be supported.
And if I recall correctly then either `androidPackage` or `iosPrefix` was optional.
In either case, if either `androidPackage` or `iosPrefix` is present then we assume `ios` and `android` support.

```dart
if (pluginMap['androidPackage'] is String || pluginMap['iosPrefix'] is String) {
  result.add(_FlutterPlatform.android);
  result.add(_FlutterPlatform.ios);
}
```

Reply:
Done - PTAL